lib/pages/events_page.dart

```diff
@@ -39,11 +58,27 @@ class _EventsPageState extends State<EventsPage> {
 
   Widget listSeperatorBuilder(_, __) => const Divider(height: 1);
 
+  Widget renderList(List<Event> list) {
```

Comment:
If this function is responsible from rendering the list, empty state ui is not its responsibility imo.
Another block of logic should decide to render the "empty state" overlay on top of the list.

Reply:
I don't know if it's necessary to have another block of logic for this. Maybe it's just the name since we render the list of events like this `eventLists.map(renderList).toList()`. But for the sake of single responsibility, I can create another function like this:
```
  Widget renderListOrShowNoEventsWidget(List<Event> list) {
    if (list.length > 0) {
      return renderList(list);
    }

    return NoEventsOverlay();
  }
```
What do you think?

Reply:
Updates: I use `AnimatedCrossFade` instead of the function above. But why we can't just keep it that simple :D We only have 2 lists.

Reply:
> But why we can't just keep it that simple :D We only have 2 lists

True :)